### PR TITLE
Remove migration logic from the stream processor

### PR DESCRIPTION
**Pull request by saig0**
## Description

* remove the migration logic from the processing/replay state machines because all records are migrated
  * remove the reprocessing stream writer because we don't reprocess commands
  * but keeping the changes minimal with respect to #7430
* rename reprocessing to replay because there is no reprocessing anymore
* align test cases to process command instead of events
  * no big overhaul of the tests with respect to #7430 

Note that the changes disable the logic to build the blacklist on replay. We should restore the behavior by #7429. Fixing it now would introduce more changes that would overlap with our next issues for building the state on the followers. 

## Related issues

contribute to #6202

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java [hunk 1]
```diff
@@ -229,23 +214,9 @@ private long scanLog(final long snapshotPosition) {
...
-
         final var recordKey = newEvent.getKey();
         // records from other partitions should not influence the key generator of this partition
         if (Protocol.decodePartitionId(recordKey) == zeebeState.getPartitionId()) {
-          lastGeneratedKeyBySourceCommandPosition.computeIfPresent(
-              sourceEventPosition, (position, key) -> Math.max(recordKey, key));
-
           // remember the highest key on the stream to restore the key generator after replay
           highestRecordKey = Math.max(recordKey, highestRecordKey);
         }
```
Comment (Zelldon) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java:223]: After merging this and https://github.com/camunda-cloud/zeebe/pull/7446 we can completely remove the scan method

**Review by ChrisKujawa — APPROVED** [review]
Awesome thanks :+1: :100:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java [hunk 2]
```diff
@@ -122,14 +122,12 @@ protected void onActorStarted() {
       healthCheckTick();
       openFuture.complete(null);
 
-      final ReProcessingStateMachine reProcessingStateMachine =
-          new ReProcessingStateMachine(processingContext);
+      final ReplayStateMachine replayStateMachine = new ReplayStateMachine(processingContext);
 
-      // disable writing to the log stream but for reprocessing checks
+      // disable writing to the log stream
       processingContext.disableLogStreamWriter();
```
[the remark this answers is not in the document]
Comment (saig0) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java:128]: No. But I would like to focus on this part in a separate PR :wink:

File: engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReProcessingStateMachine.java:223]: :thinking: yes :+1: 
We can move the additional logic (e.g. last source event position, highest record key) to the other loop.

**Pull request merged**